src/visitor/test/RewriteRelExprVisitorTest.cpp

```diff
@@ -0,0 +1,105 @@
+        ASSERT_EQ(*res, *expected) << res->toString() << " vs. " << expected->toString();
+    }
+    // (-1 + label < 40)  =>  (label < 40 - (-1))
+    {
+        auto expr =
+            pool.add(ltExpr(addExpr(constantExpr(-1), laExpr("v", "age")), constantExpr(40)));
+        auto res = pool.add(ExpressionUtils::rewriteRelExpr(expr));
+        auto expected =
+            pool.add(ltExpr(laExpr("v", "age"), minusExpr(constantExpr(40), constantExpr(-1))));
+        ASSERT_EQ(*res, *expected) << res->toString() << " vs. " << expected->toString();
+    }
+    // (label1 + label2 < 40)  =>  (label1 + label2 < 40) unchaged
+    // TODO: replace list with set in object pool and avoid copy
+    {
+        auto expr =
+            pool.add(ltExpr(addExpr(laExpr("v", "age"), laExpr("v2", "age2")), constantExpr(40)));
+        auto res = pool.add(ExpressionUtils::rewriteRelExpr(expr));
+        auto expected = expr;
+        ASSERT_EQ(*res, *expected) << res->toString() << " vs. " << expected->toString();
+    }
+}
+
+TEST_F(RewriteRelExprVisitorTest, NestedArithmeticalExpr) {
+    // (label + 1 + 2 < 40)  =>  (label < 40 - 2 - 1)
+    {
+        auto expr =
+            pool.add(ltExpr(addExpr(addExpr(laExpr("v", "age"), constantExpr(1)), constantExpr(2)),
+                            constantExpr(40)));
+        auto res = pool.add(ExpressionUtils::rewriteRelExpr(expr));
+        auto expected = pool.add(
+            ltExpr(laExpr("v", "age"),
+                   minusExpr(minusExpr(constantExpr(40), constantExpr(2)), constantExpr(1))));
+        ASSERT_EQ(*res, *expected) << res->toString() << " vs. " << expected->toString();
+    }
+    // (label + 1 - 2 < 40)  =>  (label < 40 + 2 - 1)
+    {
+        auto expr = pool.add(
+            ltExpr(minusExpr(addExpr(laExpr("v", "age"), constantExpr(1)), constantExpr(2)),
+                   constantExpr(40)));
+        auto res = pool.add(ExpressionUtils::rewriteRelExpr(expr));
+        auto expected = pool.add(
+            ltExpr(laExpr("v", "age"),
+                   minusExpr(addExpr(constantExpr(40), constantExpr(2)), constantExpr(1))));
+        ASSERT_EQ(*res, *expected) << res->toString() << " vs. " << expected->toString();
+    }
+    // (label + 1 - 2 + 3 < 40)  =>  (label < 40 - 3 + 2 - 1)
+    {
+        auto expr = pool.add(
+            ltExpr(addExpr(minusExpr(addExpr(laExpr("v", "age"), constantExpr(1)), constantExpr(2)),
+                           constantExpr(3)),
+                   constantExpr(40)));
+        auto res = pool.add(ExpressionUtils::rewriteRelExpr(expr));
+        auto expected = pool.add(
+            ltExpr(laExpr("v", "age"),
+                   minusExpr(addExpr(minusExpr(constantExpr(40), constantExpr(3)), constantExpr(2)),
+                             constantExpr(1))));
+        ASSERT_EQ(*res, *expected) << res->toString() << " vs. " << expected->toString();
+    }
+}
```

Comment:
Add nested relational expression case.

Reply:
nested relational expressions are rarely used as filters and this PR hasn't addressed this.